### ISSUE TITLE
test on the voc2007,but the map is closed to zero

### ISSUE THREAD

**Issue by nothingws** [zylo117/Yet-Another-EfficientDet-Pytorch#123]
I change the data format of XML to JSON format
for example,
[{"file_name": "000012.jpg", "height": 333, "width": 500, "id": "000012"}, {"file_name": "000017.jpg", "height": 364, "width": 480, "id": "000017"}, {"file_name": "000023.jpg", "height": 500, "width": 334, "id": "000023"}, {"file_name": "000026.jpg", "height": 333, "width": 500, "id": "000026"}, {"file_name": "000032.jpg", "height": 281, "width": 500, "id": "000032"}, 

{"segmentation": [[239, 55, 239, 332, 432, 332, 432, 55]], "area": 53461, "iscrowd": 0, "image_id": "000438", "bbox": [239, 55, 193, 277], "category_id": 15, "id": 286, "ignore": 0}, {"segmentation": [[129, 166, 129, 476, 230, 476, 230, 166]], "area": 31310, "iscrowd": 0, "image_id": "000446", "bbox": [129, 166, 101, 310], "category_id": 15, "id": 287, "ignore": 0}, {"segmentation": [[80, 85, 80, 472, 244, 472, 244, 85]], "area": 63468, "iscrowd": 0, "image_id": "000446", "bbox": [80, 85, 164, 387], "category_id": 15, "id": 288, "ignore": 0}, {"segmentation": [[37, 35, 37, 426, 325, 426, 325, 35]], "area": 112608, "iscrowd": 0, "image_id": "000450", "bbox": [37, 35, 288, 391], "category_id": 19, "id": 289, "ignore": 0},

loss
Val. Epoch: 70/500. Classification loss: 12.09009. Regression loss: 0.10500. Total loss: 12.19509
Step: 12815. Epoch: 71/500. Iteration: 178/178. Cls loss: 2.30295. Reg loss: 0.11240. Total loss: 2.41535: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:52<00:00,  1.40it/s]checkpoint...
Step: 12815. Epoch: 71/500. Iteration: 178/178. Cls loss: 2.30295. Reg loss: 0.11240. Total loss: 2.41535: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:52<00:00,  1.59it/s]
Step: 12993. Epoch: 72/500. Iteration: 178/178. Cls loss: 2.30383. Reg loss: 0.10522. Total loss: 2.40905: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:53<00:00,  1.40it/s]checkpoint...
Step: 12993. Epoch: 72/500. Iteration: 178/178. Cls loss: 2.30383. Reg loss: 0.10522. Total loss: 2.40905: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:53<00:00,  1.57it/s]
Step: 13171. Epoch: 73/500. Iteration: 178/178. Cls loss: 2.30270. Reg loss: 0.09521. Total loss: 2.39791:  99%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枆| 177/178 [01:53<00:00,  1.60it/s]checkpoint...
Step: 13171. Epoch: 73/500. Iteration: 178/178. Cls loss: 2.30270. Reg loss: 0.09521. Total loss: 2.39791: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:54<00:00,  1.56it/s]
Step: 13349. Epoch: 74/500. Iteration: 178/178. Cls loss: 2.30252. Reg loss: 0.10592. Total loss: 2.40844: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:51<00:00,  1.36it/s]checkpoint...
Step: 13349. Epoch: 74/500. Iteration: 178/178. Cls loss: 2.30252. Reg loss: 0.10592. Total loss: 2.40844: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:51<00:00,  1.59it/s]
Step: 13527. Epoch: 75/500. Iteration: 178/178. Cls loss: 2.30273. Reg loss: 0.11506. Total loss: 2.41779: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:54<00:00,  1.38it/s]checkpoint...
Step: 13527. Epoch: 75/500. Iteration: 178/178. Cls loss: 2.30273. Reg loss: 0.11506. Total loss: 2.41779: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:54<00:00,  1.55it/s]
Val. Epoch: 75/500. Classification loss: 11.72371. Regression loss: 0.10431. Total loss: 11.82802
Step: 13705. Epoch: 76/500. Iteration: 178/178. Cls loss: 2.30254. Reg loss: 0.10737. Total loss: 2.40992: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:51<00:00,  1.36it/s]checkpoint...
Step: 13705. Epoch: 76/500. Iteration: 178/178. Cls loss: 2.30254. Reg loss: 0.10737. Total loss: 2.40992: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:52<00:00,  1.59it/s]
Step: 13883. Epoch: 77/500. Iteration: 178/178. Cls loss: 2.30187. Reg loss: 0.10748. Total loss: 2.40934: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:51<00:00,  1.29it/s]checkpoint...
Step: 13883. Epoch: 77/500. Iteration: 178/178. Cls loss: 2.30187. Reg loss: 0.10748. Total loss: 2.40934: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:51<00:00,  1.59it/s]
Step: 14061. Epoch: 78/500. Iteration: 178/178. Cls loss: 2.14083. Reg loss: 0.10963. Total loss: 2.25047: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:52<00:00,  1.23it/s]checkpoint...
Step: 14061. Epoch: 78/500. Iteration: 178/178. Cls loss: 2.14083. Reg loss: 0.10963. Total loss: 2.25047: 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 178/178 [01:52<00:00,  1.58it/s]
the val  result
![微信截图_20200417142613](https://user-images.githubusercontent.com/30772337/79538715-75af1380-80b7-11ea-9a1a-99283cdf4ca7.png)
![微信截图_20200417142716](https://user-images.githubusercontent.com/30772337/79538844-b9a21880-80b7-11ea-8727-640a432bb307.png)
![微信截图_20200417142730](https://user-images.githubusercontent.com/30772337/79538852-bc9d0900-80b7-11ea-9fc9-ce8b6ed9e7d7.png)
![微信截图_20200417142744](https://user-images.githubusercontent.com/30772337/79538859-beff6300-80b7-11ea-936e-5097df6f5c44.png)
![微信截图_20200417142756](https://user-images.githubusercontent.com/30772337/79538870-c161bd00-80b7-11ea-9adb-8253b99f6230.png)
![微信截图_20200417142814](https://user-images.githubusercontent.com/30772337/79538882-c58dda80-80b7-11ea-8574-623bf1e6fc7b.png)

**Comment by zylo117** [comment]
either your dataset is corrupted or overfitting. the val loss is off the roof, didn't you notice?

**Comment by nothingws** [comment]
> either your dataset is corrupted or overfitting. the val loss is off the roof, didn't you notice?

I checked my training set label. The code changes as follows
![QQ截图20200417153405](https://user-images.githubusercontent.com/30772337/79544286-5d43f680-80c1-11ea-92c8-e4d83db09661.jpg)
I found that there was no problem with the label. Look at these pictures.
![微信截图_20200417153440](https://user-images.githubusercontent.com/30772337/79544468-a7c57300-80c1-11ea-82f8-dceea77352c5.png)
![微信截图_20200417153510](https://user-images.githubusercontent.com/30772337/79544484-abf19080-80c1-11ea-95c2-15ee6d0e9875.png)
![微信截图_20200417153525](https://user-images.githubusercontent.com/30772337/79544489-b01dae00-80c1-11ea-8f66-b1b13624ae69.png)
![微信截图_20200417153545](https://user-images.githubusercontent.com/30772337/79544501-b3189e80-80c1-11ea-9d1e-322a84fa486d.png)
![微信截图_20200417153559](https://user-images.githubusercontent.com/30772337/79544508-b57af880-80c1-11ea-88e9-2031eeb57f4d.png)
when epoch 100, the val  total loss is about 3. 3 is normal??? 

> either your dataset is corrupted or overfitting. the val loss is off the roof, didn't you notice?

**Comment by nothingws** [comment]
> either your dataset is corrupted or overfitting. the val loss is off the roof, didn't you notice?

I checked my training set label. The code changes as follows
![QQ截图20200417153405](https://user-images.githubusercontent.com/30772337/79545100-bb250e00-80c2-11ea-91cc-e247ea180bc8.jpg)
I found that there was no problem with the label. Look at these pictures.
![微信截图_20200417153440](https://user-images.githubusercontent.com/30772337/79545135-cc6e1a80-80c2-11ea-937f-be87e4417107.png)
when epoch 100, the val total loss is about 3.          3 is normal???
![微信截图_20200417153510](https://user-images.githubusercontent.com/30772337/79545153-d09a3800-80c2-11ea-8f32-e563744bf032.png)
![微信截图_20200417153525](https://user-images.githubusercontent.com/30772337/79545164-d2fc9200-80c2-11ea-88ca-9ee59f1eced5.png)
![微信截图_20200417153545](https://user-images.githubusercontent.com/30772337/79545173-d5f78280-80c2-11ea-8d77-18d4253f7678.png)
![微信截图_20200417153559](https://user-images.githubusercontent.com/30772337/79545180-d8f27300-80c2-11ea-9c7d-d94b54cafa78.png)

**Comment by nothingws** [comment]
> either your dataset is corrupted or overfitting. the val loss is off the roof, didn't you notice?

I will try on the coco that can help me to correct the error.

**Comment by gtgtgt1117** [comment]
>  I found that there was no problem with the label. Look at these pictures.




您好,我训练我的数据集检测结果和你的一样,请问您解决这个问题了吗?

**Comment by zylo117** [comment]
> > either your dataset is corrupted or overfitting. the val loss is off the roof, didn't you notice?
> 
> I will try on the coco that can help me to correct the error.

you should run the tutorial first, that will show you how to train on a custom dataset in 15 minutes.

**Comment by gtgtgt1117** [comment]
> > > either your dataset is corrupted or overfitting. the val loss is off the roof, didn't you notice?
> > 
> > 
> > I will try on the coco that can help me to correct the error.
> 
> you should run the tutorial first, that will show you how to train on a custom dataset in 15 minutes.


Hello,我是按照这个步骤来的呀,数据也没有问题,custom.yml也创建了.
python train.py -c 2 --batch_size=8 --lr=1e-3 --num_epochs 1 --load_weights=./efficientdet-d2.pth -p custom

**Comment by zylo117** [comment]
i mean THE tutorial, literally. There is a folder named tutorial. check that out.
